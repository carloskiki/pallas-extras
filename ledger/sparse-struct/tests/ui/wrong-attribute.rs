use sparse_struct::SparseStruct;

#[derive(SparseStruct)]
#[struct_name("hello")]
enum Enum {
    Variant1(u8),
    Variant2(u16),
}

#[derive(SparseStruct)]
enum Enum2 {
    Variant1(u8),
    #[struct_name("hello")]
    Variant2(u16),
}

fn main() {}
