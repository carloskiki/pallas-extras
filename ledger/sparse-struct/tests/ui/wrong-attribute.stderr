error: expected identifier
 --> tests/ui/wrong-attribute.rs:4:15
  |
4 | #[struct_name("hello")]
  |               ^^^^^^^

error: The `struct_name` should be specified on the enum, not on its variants.
  --> tests/ui/wrong-attribute.rs:13:5
   |
13 |     #[struct_name("hello")]
   |     ^
